### DISCUSSION TITLE
Force Snapshot instead of Pod Volume Backup

### DISCUSSION THREAD

**Discussion by ThomasSteinbach** [vmware-tanzu/velero#6213] [Community support Q&A] [answered]
I have Velero configured for my AWS EKS Cluster with EBS CSI driver.
However Velero seems to prefer to do a pod volume backup of EBS volumes instead of doing a snapshot.
Following is shown in the Velero logs: "Skipping snapshot of persistent volume because volume is being backed up with pod volume backup".

How can I force Velero to do snaphots of EBS volumes - and only do restic backups of my EFS volumes?

**Comment by mdnahian** [comment, marked as answer]
Hi,

It may be possible to accomplish this by defining 2 separate backups and ensuring that file system backup is not enabled for the EBS volumes either through the annotations (`backup.velero.io/backup-volumes`) or the `defaultVolumesToFsBackup` parameter.

What version of Velero are you using? Velero v1.11 made an improvement to resource filtering by adding [resource policies](https://velero.io/docs/v1.11/resource-filtering/#resource-policies) which may help in your case.

Thanks,
Md ([cloudcasa.io](https://cloudcasa.io))

**Reply by ThomasSteinbach** [reply]
I think you got me on the right path.

So Velero really is either doing a pod volume backup (preferred) or a snapshot when both is possible, right? I didn't found that behavior documented.

However I realized that Velero made snapshots from my volumes, but from Schedules of other namespaces, where I activated cluster resource backups. That means you seem to be right. As long as Velero would not do a pod volume backup, it starts doing snapshots. This behavior should be configurable :/

Resource policy could help. @mdnahian do you know how to apply resource policies to schedules? I mean the declarative API object?

**Reply by ThomasSteinbach** [reply]
Mhhh.. it seems that the only solution is indeed to have 2 separate backups - one doing snapshots and the other doing file system backups.

The resource policy filter did not work for me. I tried it with the `csi: {}` only condition as well as with the `storageClass` condition. Velero mentioned something in the logs, that the policies will apply to the resources. However Velero continues doing file system backups.

This is a bit sad, because restoring a backup with two different types of volume backups (snapshot and FSB) could not be made from a single backup.

**Comment by ThomasSteinbach** [comment]
I've found some old issues regarding this requirement:
* https://github.com/vmware-tanzu/velero/issues/2671
* https://github.com/vmware-tanzu/velero/issues/4228

**Comment by sseago** [comment]
So there is another workaround right now that's not "do two separate backups" -- if you're not defaulting to PodVolumeBackup, then you can annotate your pods that use EFS volumes with the pvb annotation (see file system backup documentation). This is not as convenient as "do podvolumebackups when csi isn't supported", but it may be more convenient than doing two backups. This is something considered for a future feature -- basically augmenting the resource policies feature to allow something like this based on volume type, but it does not (yet) exist.

**Reply by draghuram** [reply]
@sseago, if I have a backup with "snapshotVolumes" set to "true" and if I have PVB annotations, the backup should do both snapshots and file system backup?

**Reply by sseago** [reply]
@draghuram no, Velero will never do both. If you have PVB annotations, then kopia/restic will back up the volume rather than using snapshots.
